packages/ods/src/components/quantity/src/components/ods-quantity/ods-quantity.tsx

```diff
@@ -0,0 +1,146 @@
+  @Prop({ mutable: true, reflect: true }) public value: number | null = null;
+
+  @Event() odsBlur!: EventEmitter<void>;
+  @Event() odsChange!: EventEmitter<OdsQuantityEventValueChangeDetail>;
+  @Event() odsClear!: EventEmitter<void>;
+  @Event() odsFocus!: EventEmitter<void>;
+  @Event() odsReset!: EventEmitter<void>;
+
+  @Method()
+  async clear(): Promise<void> {
+    return this.odsInput?.clear();
+  }
+
+  @Method()
+  async getValidity(): Promise<ValidityState | undefined> {
+    return this.odsInput?.getValidity();
+  }
+
+  @Method()
+  async reset(): Promise<void> {
+    return this.odsInput?.reset();
+  }
+
+  componentWillLoad(): void {
+    if (!this.value && this.value !== 0) {
+      this.value = this.defaultValue ?? null;
+    }
+    setFormValue(this.internals, this.value);
+  }
+
+  async formResetCallback(): Promise<void> {
+    await this.reset();
+  }
+
+  private decrement(): void {
+    if (this.isDisabled || this.isReadonly) {
+      return;
+    }
+    const step = this.step || 1;
+    this.value = this.value !== null ? Number(this.value) - step : 0;
+  }
+
+  private increment(): void {
+    if (this.isDisabled || this.isReadonly) {
+      return;
+    }
+    const step = this.step || 1;
+    this.value = this.value !== null ? Number(this.value) + step : 0;
+  }
+
+  private onOdsChange(event: OdsInputValueChangeEvent): void {
+    if (event.detail.value === null) {
+      this.value = null;
+      return;
+    }
+    this.value = Number(event.detail.value) ?? null;
+    setFormValue(this.internals, this.value);
+  }
+
+  render(): FunctionalComponent {
+    return (
+      <Host class="ods-quantity">
+        <ods-button
+          class={{
+            'ods-quantity__button': true,
+            'ods-quantity__button--readonly': this.isReadonly,
+          }}
+          color={ this.hasError ? ODS_BUTTON_COLOR.critical : ODS_BUTTON_COLOR.primary }
+          exportparts="button:button_minus"
+          isDisabled={ isMinusButtonDisabled(this.isDisabled, this.value, this.min) }
+          icon={ ODS_ICON_NAME.minus }
+          label=""
+          onClick={ () => this.decrement() }
+          onFocus={ () => this.odsFocus.emit() }
+          size={ ODS_BUTTON_SIZE.sm }
+          variant={ ODS_BUTTON_VARIANT.outline }>
+        </ods-button>
+        <ods-input
+          ariaLabel={ this.ariaLabel }
+          ariaLabelledby= { this.ariaLabelledby }
+          class="ods-quantity__input"
+          defaultValue={ this.defaultValue }
+          exportparts="input"
+          hasError={ this.hasError }
+          isDisabled={ this.isDisabled }
+          isReadonly={ this.isReadonly }
+          isRequired={ this.isRequired }
+          onOdsChange={ (event: OdsInputValueChangeEvent) => this.onOdsChange(event) }
+          max={ this.max }
+          min={ this.min }
+          name={ this.name }
+          placeholder={ this.placeholder }
+          ref={ (el?: unknown) => this.odsInput = el as OdsInput }
```

Comment:
use `HTMLOdsInputElement` instead? (by changing also odsInput type to HTMLOdsInputElement

Reply:
Don't work on the CI test